src/lib.rs

```diff
@@ -237,6 +237,152 @@ impl<T: Clone + Integer> Ratio<T> {
     pub fn fract(&self) -> Ratio<T> {
         Ratio::new_raw(self.numer.clone() % self.denom.clone(), self.denom.clone())
     }
+
+    #[cfg(feature = "std")]
+    pub fn as_decimal_string(&self, precision: usize, trail_zeroes: bool) -> std::string::String
+    where
+        T: CheckedMul + fmt::Display
+    {
+        use std::string::String;
+
+        let int_len = {
+            // Build the number 10 from scratch so we can handle both signed and unsigned types (T of i8 or u8)
+            let mut _10 = T::one() + T::one() + T::one(); // 3
+            _10 = _10.clone() * _10 + T::one();           // 10
+
+            let mut ptr: T = T::one();
+            let mut len: usize = 0;
+
+            loop {
+                len += 1;
+
+                ptr = if let Some(p) = ptr.checked_mul(&_10) { p } else { break };
+
+                if ptr > *self.numer() {
+                    break;
+                }
+            }
+
+            len
+        };
+
+        let mut result = String::with_capacity(int_len + precision + if precision > 0 { 1 } else { 0 });
+
+        self.format_as_decimal(&mut result, precision, trail_zeroes).ok();
+
+        result
+    }
+
+    /// Writes the number into the output buffer in decimal representation.  
+    /// No leading zeroes (except a single zero for the integer part)  
+    /// Returns remainder of the division
```

Comment:
Oh, I meant to comment here too -- it's not really clear what this remainder represents. It's not just `numer % denom`, but rather `numer % (denom * 10^precision)`, right? I'm not really sure what anyone would do with that, except check that it's non-zero as you have in tests. That gets hairier if we start rounding too...

Reply:
No wait, it must be `(numer * 10^precision) % denom`, right? See, this is not obvious... :)